### DISCUSSION TITLE
Unusual Ca+ imaging data -help/guidance for NWB conversion

### DISCUSSION THREAD

**Discussion by ucscbrianlee** [NeurodataWithoutBorders/helpdesk#77] [Q&A] [answered]
Dear NWB,

Our consortium has a dataset that is in preprint (https://www.biorxiv.org/content/10.1101/2023.11.30.569494v1) that features an unusual Ca+ imaging approach.  The cell mask is from a different experiment, that is to say the segmentation is not derived from the raw Ca+ imaging files, thus the analysis is a combination of the raw files and a rotated/processed/cropped cell mask from a different MERFISH experiment. 

The paper demonstrates a kind of new Pertrub-Seq + MERFISH (Multiplexed Error-Robust Fluorescence In Situ Hybridization) approach called Perturb-FISH.   The spatial transcriptomics approach of MERFISH is combined with the CRISPR-mediated gene perturbations seen in Perturb-Seq to modify genomes. Counting the number of fluorescent RNA targets labeled by fluorescently tagged oligo probes, the MERFISH component determines the expression profile of each gene, as well as identifies which cells have been perturbed.   MERFISH uses this raw data to build a cell mask for all of this analysis. Ca+ imaging data was collected to help confirm the pertubations were real in terms of expected phenotypes and to also gain spatial information, in regards to which active cells might relate to nearby inactive/active cells. 

The Ca+ imaging data was conducted on the same biosample so that the same spatial arrangement persists in the MERFISH and Ca+ imaging experiment.  Changes in fluorescent intensity in regard to calcium levels were recorded with individual TIFF files taken over a set time intervals (different than the approach of most Ca+ imaging commercial devices -in that instead of a movie there is a directory with many TIFF files of individual time captures). The cell mask from the MERFISH experiment (output from Merlin software) was rotated, cropped and aligned with the small field of view calcium experiment.  This required some computational steps and then with the cell identities labeled the raw Ca+ imaging could track optical electrophysiology for each cell. 

This data is aimed for the DANDI repository, requiring the NWB format.  Some questions: can the lab just overlay the raw TIFF data with the not-so-aligned cell mask from Merlin and and include some transformation information so that anyone using the data later can decouple the individual parts and recreate the final analysis which also needs transforming? 

Any suggested guidance is welcomed and eagerly sought.  Also many apologies for any mischaracterizations I may be doing to the data and experimental steps.

Brian Lee

**Comment by oruebel** [comment]
Thanks for reaching out. @bendichter @rly could you take a look at this thread.

**Comment by bendichter** [comment]
Hi Brian,

The `PlaneSegmentation` table does not really assume any particular workflow, so I don't see why you couldn't use it here. It does sort of *imply* something like CaImAn or suite2p, but does not require it. I would recommend using the standard workflow outlined in the PyNWB Calcium Imaging tutorial, and simply put a brief description of your methodology in the description of the `PlaneSegmentation` table.

**Comment by ucscbrianlee** [comment, marked as answer]
@bendichter Great!  Thank you for sharing. I'll bring this back to the lab and let them know it should be OK to layer the data as they have it with some methodology description to help future data users understand the provenance and structure.
